From d14a838d7d0ef1b009b2bcc46e4908dc8a3daf62 Mon Sep 17 00:00:00 2001
From: Golan <gaharon@habana.ai>
Date: Sun, 02 Aug 2020 12:28:17 +0300
Subject: [PATCH] sve:build (SVE using buildroot-external support)

Running sve (silicon validation embedded) to validated HW features with
minimal embedded suuport.

no need for new varabiles (uses buildroot-external varaibles)

e.g. to build sve
>make sve-rebuild

e.g to clean sve
>make sve-dirclean

on suddessfull build bin file can be found @
output/$(HL_DEV)/build/sve-custom/sve.bin

Change-Id: Ib21da90bb0371a99abd83a66050c8ee5948390c6
Signed-off-by: Golan <gaharon@habana.ai>
---

diff --git a/Config.in b/Config.in
index 8af0698..030b785 100644
--- a/Config.in
+++ b/Config.in
@@ -1,5 +1,6 @@
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/armcp/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/dsv/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/sve/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/thermal/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/preboot/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/btl/Config.in"
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index d9583b6..4937d04 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -2,6 +2,7 @@
 LINUX_OVERRIDE_SRCDIR = $(TOPDIR)/../linux
 ARMCP_OVERRIDE_SRCDIR = $(TOPDIR)/../armcp
 DSV_OVERRIDE_SRCDIR = $(TOPDIR)/../dsv
+SVE_OVERRIDE_SRCDIR = $(TOPDIR)/../sival_embedded
 THERMAL_OVERRIDE_SRCDIR = $(TOPDIR)/../thermal
 PREBOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
 HL_FW_LOADER_OVERRIDE_SRCDIR = $(TOPDIR)/../fw_loader
diff --git a/board/habanalabs/common/sve.its b/board/habanalabs/common/sve.its
new file mode 100644
index 0000000..c9a3681
--- /dev/null
+++ b/board/habanalabs/common/sve.its
@@ -0,0 +1,37 @@
+/*
+ * Simple standalone image source file containing a
+ * single HabanaLabs SVE app
+ */
+
+/dts-v1/;
+
+/ {
+	hl-device = "goya";
+	description = "FIT SVE image";
+	#address-cells = <2>;
+
+	images {
+		sve {
+			description = "HabanaLabs SVE (64-bit)";
+			data = /incbin/("sve.bin");
+			type = "standalone";
+			arch = "arm64";
+			os = "u-boot";
+			compression = "none";
+			load = <0x7F 0xE0000000>;
+			hash {
+				algo = "sha256";
+			};
+		};
+	};
+
+	configurations {
+		default = "fw-loader-conf";
+		fw-loader-conf {
+			description = "Habanalabs SVE";
+			loadables = "sve";
+			load-bootloader;
+			load-type = <0>; /* FIT_LOAD_TYPE_RUN */
+		};
+	};
+};
diff --git a/configs/goya_defconfig b/configs/goya_defconfig
index 9a8972f..21e0979 100644
--- a/configs/goya_defconfig
+++ b/configs/goya_defconfig
@@ -59,6 +59,7 @@
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SIGNATURE_SUPPORT=y
 BR2_PACKAGE_DSV=y
+BR2_PACKAGE_SVE=y
 BR2_PACKAGE_THERMAL=y
 BR2_PACKAGE_PREBOOT=y
 BR2_PACKAGE_BTL=y
diff --git a/package/sve/Config.in b/package/sve/Config.in
new file mode 100644
index 0000000..65d193c
--- /dev/null
+++ b/package/sve/Config.in
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_SVE
+	bool "SVE"
+	help
+	  Silicon Validation Embedded (SVE) is a tool that will run from U-Boot
diff --git a/package/sve/sve.mk b/package/sve/sve.mk
new file mode 100644
index 0000000..7d18451
--- /dev/null
+++ b/package/sve/sve.mk
@@ -0,0 +1,9 @@
+SVE_VERSION = 1.0.0
+SVE_SITE = sve
+SVE_LICENSE = unclear
+
+define SVE_BUILD_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
+endef
+
+$(eval $(generic-package))
diff --git a/scripts/make/common/Makefile b/scripts/make/common/Makefile
index 56981dc..b488894 100644
--- a/scripts/make/common/Makefile
+++ b/scripts/make/common/Makefile
@@ -200,6 +200,7 @@
 UBOOT_BASE := ${OUTPUT_DIR}/build/uboot-custom
 LINUX_BASE := ${OUTPUT_DIR}/build/linux-custom
 DSV_BASE := ${OUTPUT_DIR}/build/dsv-custom
+SVE_BASE := ${OUTPUT_DIR}/build/sve-custom
 
 HL_RELEASE_VERS_FILE ?= hl-release-version
 
